subsys/bluetooth/host/adv.c

```diff
@@ -1363,6 +1363,13 @@ int bt_le_adv_start(const struct bt_le_adv_param *param,
 	struct bt_le_ext_adv *adv = adv_get_legacy();
 	int err;
 
+	if (!(param->options & BT_LE_ADV_OPT_ONE_TIME) &&
+	    (param->options & BT_LE_ADV_OPT_CONNECTABLE) && !param->peer) {
+		LOG_WRN("Deprecation warning: %s BT_LE_ADV_OPT_CONNECTABLE without "
+			"BT_LE_ADV_OPT_ONE_TIME",
+			__func__);
```

Comment:
Do we need runtime warnings after buildtime warnings?

Reply:
The runtime warning is for apps that don't use `BT_LE_ADV_CONN`, but use a custom `bt_le_adv_param`.

Reply:
Your arguments from https://github.com/zephyrproject-rtos/zephyr/pull/73395#discussion_r1617674980 w.r.t. 

> Reusing BT_LE_ADV_CONN ever introduces the hazard of old code compiling just fine, and even running apparently just fine at first, except eventually discovering that reconnect does not work.

Also applies here IMO. If we want compile-time checks, we need to deprecate the macro and add a new one. 

Ideally all deprecation of macros such as what we do here, needs to follow the steps in https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#introducing-breaking-api-changes and the architecture group may have some opinions as well.